docs/utils/masked.rst

```diff
@@ -0,0 +1,136 @@
+.. |Masked| replace:: :class:`~astropy.utils.masked.Masked`
+.. |Quantity| replace:: :class:`~astropy.units.Quantity`
+.. _utils-masked:
+
+**************************************
+Masked Values (`astropy.utils.masked`)
+**************************************
+
+Often, data sets are incomplete or corrupted and it would be handy to be able
+to mask certain values.  Astropy provides a |Masked| class to help represent
+such data sets.
+
+.. warning:: |Masked| is experimental! While we hope basic usage will remain
+   similar, we are not yet sure whether it will not be necessary to change it
+   to make things work throughout Astropy. This also means that comments and
+   suggestions for improvements are especially welcome!
+
+.. note:: |Masked| is similar to Numpy's :class:`~numpy.ma.MaskedArray`,
+   but it supports subclasses much better and also has some important
+   :ref:`differences in behaviour <utils-masked-vs-numpy-maskedarray>`.
+
+Usage
+=====
+
+Astropy |Masked| instances behave like `~numpy.ndarray` or subclasses such as
+|Quantity| but with a mask associated, which is propagated in operations such
+as addition, etc.::
+
+  >>> import numpy as np
+  >>> from astropy import units as u
+  >>> from astropy.utils.masked import Masked
+  >>> ma = Masked([1., 2., 3.], mask=[False, False, True])
+  >>> ma
+  MaskedNDArray([1.0, 2.0, ———])
+  >>> mq = ma * u.m
+  >>> mq + 25 * u.cm
+  <MaskedQuantity [1.25, 2.25,  ———] m>
+
+You can get the values without the mask using
+`~astropy.utils.masked.Masked.unmasked`, or, if you need to control what
+should be substituted for any masked values, with
+:meth:`~astropy.utils.masked.Masked.unmask`::
+
+  >>> mq.unmasked
```

Comment:
Yes, I tried to have the (very limited) API to have "mask" in the names to avoid overloading. Here, I particularly liked that I could then have a matching `unmask()` method.

But as you will see from the initializer, "data" and "mask" is definitely in my mind too...